ex3/server/changeAlgoSettings.h

```diff
@@ -0,0 +1,67 @@
+#include "Command.h"
+#include "KnnClassifier.h"
+#include "../utils.h"
+#include "stdexcept"
+#include "ChebyshevDistance.h"
+#include "EuclideanDistance.h"
+#include "ManhattanDistance.h"
+
+#ifndef CLIENT_CHANGEALGOSETTINGS_H
+#define CLIENT_CHANGEALGOSETTINGS_H
+
+template<class T>
+class changeAlgoSettings : public Command {
+private:
+    KnnClassifier<T> *algo;
+public:
+    void execute() override {
+        this->getDefaultIO()->write(this->algo->toString());
+        std::string input = this->getDefaultIO()->read();
+        if (input.empty()) {
+            this->getDefaultIO()->write(this->algo->toString());
+            return;
+        }
+        std::vector<std::string> parameters = utils::split(input, ' ');
+        if (parameters.size() != 2) {
+            this->getDefaultIO()->write("Expected 2 parameters");
+            return;
+        }
+        int k;
+        try {
+            k = std::stoi(parameters[0]);
+        }
+        catch (const std::invalid_argument &exception) {
+            this->getDefaultIO()->write("K should be an int");
+            return;
+        }
+
+        if (k > 10 || k < 1) {
+            this->getDefaultIO()->write("K should be between 1 and 10");
+            return;
+        }
+        Distance<T> *newDistance;
+        ChebyshevDistance<T> chebyshevDistance;
+        ManhattanDistance<T> manhattanDistance;
+        EuclideanDistance<T> euclideanDistance;
+        std::string inputDistance = parameters[1];
+        if (inputDistance == "CHE" || inputDistance == "che") {
+            newDistance = &chebyshevDistance;
+        } else if (inputDistance == "MAN" || inputDistance == "man") {
+            newDistance = &manhattanDistance;
+        } else if (inputDistance == "EUC" || inputDistance == "euc") {
+            newDistance = &euclideanDistance;
```

Comment:
Turn inputDistance into lowercase and then compare:
https://stackoverflow.com/questions/313970/how-to-convert-an-instance-of-stdstring-to-lower-case